xmtp_id/src/associations/serialization.rs

```diff
@@ -300,6 +342,30 @@ fn to_member_identifier_proto(member_identifier: MemberIdentifier) -> MemberIden
     }
 }
 
+fn to_association_state_proto(association_state: AssociationState) -> AssociationStateProto {
```

Comment:
can this be a `impl From<AssociationState> for AssociationStateProto`?

Reply:
Yeah this is implemented below

The reason I split this up into functions is to follow the precedent already set in the serialization file. although I would prefer to only use `TryFrom` and `From` more generally. I implemented `From`/`TryFrom` in places where my code used it elsewhere

I think we should create a ticket for converting these standalone functions into their own `From`/`TryFrom` unless there is a good reason for keeping them as standalone. It's an easier task but would be nice not to make a different ticket too noisy. 

@neekolas @richardhuaaa WDYT of converting these into trait implementations?

Reply:
One consideration I had when designing it this way is the likelihood that we are going to need to pass in some sort of `ERC1271SignatureVerifier` in to these functions so that we have everything we need to verify the signatures on the `IdentityUpdate`.

That struct is going to need to hold the RPC URL etc and can't just be a singleton.

Reply:
We can have a mapping of CAIP-10 to RPC URL and keep it in libxmtp, so that we don't need to de/serialize them.

Reply:
Opened a new [pr](https://github.com/xmtp/libxmtp/pull/672) to change some of these standalones into the trait where possible